animations/shared-element-transitions/index.md

```diff
@@ -0,0 +1,684 @@
+---
+layout: page
+title: "Shared Element Transitions"
+permalink: /animations/shared-element-transitions/
+---
+
+<div class="whats-the-point" markdown="1">
+
+<b> <a id="whats-the-point" class="anchor" href="#whats-the-point" aria-hidden="true"><span class="octicon octicon-link"></span></a>What you'll learn:</b>
+
+* Create a shared element transition using Flutter's Hero widget.
+* Fly the Hero's child from one page to another.
+* Animate the transformation of a Hero's shape from circular to
+  rectangular, while flying it from one page to another.
+</div>
+
+This guide demonstrates how to build shared element transitions
+in Flutter, also called shared element _animations_. Flutter
+calls these _Hero animations_ because they are implemented
+with Hero widgets.
+
+* TOC Placeholder
+{:toc}
+
+<aside class="alert alert-info" markdown="1">
+**New to Flutter?**
+You should first understand how to create a layout using Flutter’s
+widgets.  For more information, see [Building Layouts in
+Flutter](/tutorials/layout/).
+</aside>
+
+<aside class="alert alert-info" markdown="1">
+**Terminology:**
+A _Route_ describes a page or screen in a Flutter app. These terms
+are sometimes used interchangeably.
+</aside>
+
+A shared element transition animates a Hero's child
+(typically an image) from one route to another.
+This guide focuses on two types of Hero animations:
+
+<ul>
+<li markdown="1">Flying transitions<br>
+A _flying transition_ animates the Hero
+from one route to a new route, usually landing at a different
+location and with a different size.
+
+The following video (recorded at slow speed) shows a typical example.
+Tapping the flippers in the center of the screen flies them to the
+upper left corner of a new, blue screen, at a smaller size.
+Tapping the flippers in the blue screen
+(or using the device's back-to-previous-screen gesture)
+flies the flippers back to the original screen.
+
+<center><iframe width="560" height="315" src="https://www.youtube.com/embed/uF7_JfhBsww" frameborder="0" allowfullscreen></iframe></center>
```

Comment:
Shams, can you add '&rel=0' to the end of the YouTube url to prevent distracting related videos from showing up after the demo video finishes? I found this trick here: https://wptavern.com/how-to-disable-related-videos-for-youtube-embeds-in-wordpress.

Reply:
OMG! What a wonderful tip!!!

Reply:
Oh bummer. Adding &rel=0 to the URL breaks the video.